Plutarch/Api/V1/Tx.hs

```diff
@@ -38,6 +43,24 @@ instance DerivePlutusType PTxId where type DPTStrat _ = PlutusTypeData
 instance PUnsafeLiftDecl PTxId where type PLifted PTxId = Plutus.TxId
 deriving via (DerivePConstantViaData Plutus.TxId PTxId) instance PConstantDecl Plutus.TxId
 
+instance PTryFrom PData PTxId where
+  type PTryFromExcess PData PTxId = Flip Term PTxId
+  ptryFrom' opq cont = ptryFrom @(PAsData PTxId) opq (cont . first punsafeCoerce)
+
+instance PTryFrom PData (PAsData PTxId) where
+  type PTryFromExcess PData (PAsData PTxId) = Flip Term PTxId
+  ptryFrom' opq = runTermCont $ do
+    opq' <- tcont . plet $ pasConstr # opq
+    dataBs <- tcont $ \f ->
+      pif
+        (pfstBuiltin # opq' #== 0 #&& plength # (psndBuiltin # opq') #== 1)
```

Comment:
This is inefficient. You're traversing the list twice.

Reply:
Optimally you should be able to first verify it's a `PDataSum`, but because it doesn't return any excess, it's not efficient.

Reply:
This seems like a big drawback, why doesn't `PDataSum` return an excess with fields?

Reply:
(Human) laziness. I'm sure there's a way to do it, but I'd rather rethink the whole excess field bullshit.